File: tests/algorithms/x/PHP/physics/newtons_law_of_gravitation.php
<?php
error_reporting(E_ALL & ~E_DEPRECATED);
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _panic($msg) {
    fwrite(STDERR, strval($msg));
    exit(1);
}
$__start_mem = memory_get_usage();
$__start = _now();
  $GRAVITATIONAL_CONSTANT = 0.000000000066743;
  function sqrtApprox($x) {
  global $GRAVITATIONAL_CONSTANT, $r1, $r2, $r3, $r4;
  $guess = $x / 2.0;
  $i = 0;
  while ($i < 20) {
  $guess = ($guess + $x / $guess) / 2.0;
  $i = $i + 1;
};
  return $guess;
};
  function gravitational_law($force, $mass_1, $mass_2, $distance) {
  global $GRAVITATIONAL_CONSTANT, $r1, $r2, $r3, $r4;
  $zero_count = 0;
  if ($force == 0.0) {
  $zero_count = $zero_count + 1;
}
  if ($mass_1 == 0.0) {
  $zero_count = $zero_count + 1;
}
  if ($mass_2 == 0.0) {
  $zero_count = $zero_count + 1;
}
  if ($distance == 0.0) {
  $zero_count = $zero_count + 1;
}
  if ($zero_count != 1) {
  _panic('One and only one argument must be 0');
}
  if ($force < 0.0) {
  _panic('Gravitational force can not be negative');
}
  if ($distance < 0.0) {
  _panic('Distance can not be negative');
}
  if ($mass_1 < 0.0) {
  _panic('Mass can not be negative');
}
  if ($mass_2 < 0.0) {
  _panic('Mass can not be negative');
}
  $product_of_mass = $mass_1 * $mass_2;
  if ($force == 0.0) {
  $f = $GRAVITATIONAL_CONSTANT * $product_of_mass / ($distance * $distance);
  return ['kind' => 'force', 'value' => $f];
}
  if ($mass_1 == 0.0) {
  $m1 = $force * ($distance * $distance) / ($GRAVITATIONAL_CONSTANT * $mass_2);
  return ['kind' => 'mass_1', 'value' => $m1];
}
  if ($mass_2 == 0.0) {
  $m2 = $force * ($distance * $distance) / ($GRAVITATIONAL_CONSTANT * $mass_1);
  return ['kind' => 'mass_2', 'value' => $m2];
}
  $d = sqrtApprox($GRAVITATIONAL_CONSTANT * $product_of_mass / $force);
  return ['kind' => 'distance', 'value' => $d];
};
  $r1 = gravitational_law(0.0, 5.0, 10.0, 20.0);
  $r2 = gravitational_law(7367.382, 0.0, 74.0, 3048.0);
  $r3 = gravitational_law(100.0, 5.0, 0.0, 3.0);
  $r4 = gravitational_law(100.0, 5.0, 10.0, 0.0);
  echo rtrim($r1['kind'] . ' ' . _str($r1['value'])), PHP_EOL;
  echo rtrim($r2['kind'] . ' ' . _str($r2['value'])), PHP_EOL;
  echo rtrim($r3['kind'] . ' ' . _str($r3['value'])), PHP_EOL;
  echo rtrim($r4['kind'] . ' ' . _str($r4['value'])), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage(true);
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/physics/newtons_law_of_gravitation.fs
// Generated 2025-08-09 23:14 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec _str v =
    let s = sprintf "%A" v
    let s = if s.EndsWith(".0") then s.Substring(0, s.Length - 2) else s
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
type Result = {
    mutable _kind: string
    mutable _value: float
}
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let GRAVITATIONAL_CONSTANT: float = 0.000000000066743
let rec sqrtApprox (x: float) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable x = x
    try
        let mutable guess: float = x / 2.0
        let mutable i: int = 0
        while i < 20 do
            guess <- (guess + (x / guess)) / 2.0
            i <- i + 1
        __ret <- guess
        raise Return
        __ret
    with
        | Return -> __ret
let rec gravitational_law (force: float) (mass_1: float) (mass_2: float) (distance: float) =
    let mutable __ret : Result = Unchecked.defaultof<Result>
    let mutable force = force
    let mutable mass_1 = mass_1
    let mutable mass_2 = mass_2
    let mutable distance = distance
    try
        let mutable zero_count: int = 0
        if force = 0.0 then
            zero_count <- zero_count + 1
        if mass_1 = 0.0 then
            zero_count <- zero_count + 1
        if mass_2 = 0.0 then
            zero_count <- zero_count + 1
        if distance = 0.0 then
            zero_count <- zero_count + 1
        if zero_count <> 1 then
            failwith ("One and only one argument must be 0")
        if force < 0.0 then
            failwith ("Gravitational force can not be negative")
        if distance < 0.0 then
            failwith ("Distance can not be negative")
        if mass_1 < 0.0 then
            failwith ("Mass can not be negative")
        if mass_2 < 0.0 then
            failwith ("Mass can not be negative")
        let product_of_mass: float = mass_1 * mass_2
        if force = 0.0 then
            let f: float = (GRAVITATIONAL_CONSTANT * product_of_mass) / (distance * distance)
            __ret <- { _kind = "force"; _value = f }
            raise Return
        if mass_1 = 0.0 then
            let m1: float = (force * (distance * distance)) / (GRAVITATIONAL_CONSTANT * mass_2)
            __ret <- { _kind = "mass_1"; _value = m1 }
            raise Return
        if mass_2 = 0.0 then
            let m2: float = (force * (distance * distance)) / (GRAVITATIONAL_CONSTANT * mass_1)
            __ret <- { _kind = "mass_2"; _value = m2 }
            raise Return
        let d: float = sqrtApprox ((GRAVITATIONAL_CONSTANT * product_of_mass) / force)
        __ret <- { _kind = "distance"; _value = d }
        raise Return
        __ret
    with
        | Return -> __ret
let r1: Result = gravitational_law (0.0) (5.0) (10.0) (20.0)
let r2: Result = gravitational_law (7367.382) (0.0) (74.0) (3048.0)
let r3: Result = gravitational_law (100.0) (5.0) (0.0) (3.0)
let r4: Result = gravitational_law (100.0) (5.0) (10.0) (0.0)
printfn "%s" (((r1._kind) + " ") + (_str (r1._value)))
printfn "%s" (((r2._kind) + " ") + (_str (r2._value)))
printfn "%s" (((r3._kind) + " ") + (_str (r3._value)))
printfn "%s" (((r4._kind) + " ") + (_str (r4._value)))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
